check/src/kindcheck.rs

```diff
@@ -171,6 +171,7 @@ impl<'a> KindCheck<'a> {
             }
             Type::Variable(_) => Ok((self.subs.new_var(), typ.clone())),
             Type::Builtin(builtin_typ) => Ok((self.builtin_kind(builtin_typ), typ.clone())),
+            Type::Forall(_, _) => unimplemented!(),
```

Comment:
Shouldn't it be the kind of the instantiated type? `ForAll(vars, typ) => instantiate(vars, typ).kind()`?